### PR TITLE
stats: fix bug when merge sample collectors

### PR DESCRIPTION
**Pull request by lamxTyler**
`Count` should be the total number of non-null rows, not sampled rows.
PTAL @coocood @hanfei1991 @winoros

### CODE REVIEW

File: statistics/sample_test.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (s *testSampleSuite) TestMergeSampleCollector(c *C) {
 	c.Assert(len(collectors), Equals, 2)
 	collectors[0].MergeSampleCollector(collectors[1])
 	c.Assert(collectors[0].Sketch.NDV(), Equals, int64(9280))
-	c.Assert(len(collectors[0].Samples), Equals, 10000)
+	c.Assert(len(collectors[0].Samples), Equals, 1000)
```
Comment (coocood) [inline statistics/sample_test.go]: The change of the test doesn't look like covering this bug.

Comment (lamxTyler) [reply statistics/sample_test.go]: It covers, origin code will make the `Count` at line 112 not equal to `19000`

**Comment by coocood** [comment]
The `seenValues ` and `Count` is confusing, how about define a new type `SampleMerger` which contains `SampleCollector`, `TotalCount` and `TotalNullCount` field?

**Comment by coocood** [comment]
Add a field `isMerger` in `SampleCollector`, then remove argument `insertSketch` would be more clear.

**Comment by coocood** [comment]
LGTM

**Pull request merged**